yaptide/admin/simulators.py

```diff
@@ -42,437 +26,210 @@ class SimulatorType(IntEnum):
 topas_bucket_name = os.getenv('S3_TOPAS_BUCKET')
 topas_key = os.getenv('S3_TOPAS_KEY')
 topas_version = os.getenv('S3_TOPAS_VERSION')
-geant_bucket_name = os.getenv('S3_GEANT_BUCKET')
+geant4_bucket_name = os.getenv('S3_GEANT4_BUCKET')
+fluka_bucket = os.getenv('S3_FLUKA_BUCKET')
+fluka_key = os.getenv('S3_FLUKA_KEY')
 
 
 @click.group()
 def run():
     """Manage simulators"""
 
 
-@run.command
-def installed(**kwargs):
-    """List installed simulators"""
-    click.echo('to be implemented')
+def s3credentials(required: bool = False):
+    """Collection of options for S3 credentials"""
+
+    def decorator(func):
+        """Decorator for S3 credentials options"""
+        func = click.option('--endpoint',
+                            type=click.STRING,
+                            required=required,
+                            envvar='S3_ENDPOINT',
+                            default=endpoint,
+                            help='S3 endpoint')(func)
+        func = click.option('--access_key',
+                            type=click.STRING,
+                            required=required,
+                            envvar='S3_ACCESS_KEY',
+                            default=access_key,
+                            help='S3 access key')(func)
+        func = click.option('--secret_key',
+                            type=click.STRING,
+                            required=required,
+                            envvar='S3_SECRET_KEY',
+                            default=secret_key,
+                            help='S3 secret key')(func)
+        return func
+
+    return decorator
+
+
+def encryption_options(required: bool = False):
+    """Collection of options for encryption"""
+
+    def decorator(func):
+        """Decorator for encryption options"""
+        func = click.option('--password',
+                            type=click.STRING,
+                            envvar='S3_ENCRYPTION_PASSWORD',
+                            default=password,
+                            required=required,
+                            help='encryption password')(func)
+        func = click.option('--salt',
+                            type=click.STRING,
+                            envvar='S3_ENCRYPTION_SALT',
+                            default=password,
+                            required=required,
+                            help='encryption salt')(func)
+        return func
+
+    return decorator
```

Comment:
I like this approach.

Also, this can be rewritten as:

```suggestion
def encryption_options(required: bool = False):
    """Collection of options for encryption"""

    def decorator(func):
        """Decorator for encryption options"""

        @click.option('--password',
                      type=click.STRING,
                      envvar='S3_ENCRYPTION_PASSWORD',
                      default=password,
                      required=required,
                      help='encryption password')
        @click.option('--salt',
                      type=click.STRING,
                      envvar='S3_ENCRYPTION_SALT',
                      default=password,
                      required=required,
                      help='encryption salt')
        @wraps(func)
        def wrapper(*args, **kwargs):
            return func(*args, **kwargs)
        
        return wrapper

    return decorator
```

and with importing `from functools import wraps`


Same applies to `s3credentials` decorator.
Choose one that you like better

More about `@wrap` https://www.geeksforgeeks.org/python-functools-wraps-function/

Reply:
Indeed wrap is more elegant, adapted it in 21a809a8cca4e9c8a69ad4d875e4b8dacbb3d620